Versioning_Engine/Convert/ToNewVersion.cs

```diff
@@ -91,7 +91,12 @@ public static BsonDocument ToNewVersion(this BsonDocument document, string versi
                 // Get the new version back
                 BsonDocument result = ReadDocument(pipe);
                 if (result != null)
-                    document = result;
+                {
+                    if (result.Contains("_t") && result["_t"] == "NoUpdate" && result.Contains("Message"))
```

Comment:
Maybe a dumb question from not having read in detail everything on the Versioning_TK PR, but if the result contains `_t` and that type is of `NoUpdate` but the result does not contain a message, do we still want to set the `document = result;`?

Reply:
Thanks @FraserGreenroyd, Good catch. That should never happen since we are responsible for both end of the pipeline but better safe than sorry 😄 . Code updated